benchmarks/clients.rb

```diff
@@ -57,7 +57,7 @@
         count = 0
         error = 0
 
-        conn = EventMachine::HttpRequest.new(url)
+        conn = EventMachine::AblyHttpRequest::HttpRequest.new(url)
```

Comment:
Although I am not comfortable with renaming some of those classes `AblyHttpRequest`. Since, future merging of related bug fixes / feature releases will generate tons of conflicts.

I am hoping those issues will be fixed in the original library so we can switch back to the same.

Reply:
Also, we will anyway need to use exact/locked version of this library in `ably-ruby` to make sure it works as expected for provided version. Otherwise, including relevant missing tests will be super useful.

Reply:
> Although I am not comfortable with renaming to `AblyHttpRequest`. Since, future merging of related bug fixes / feature releases will generate tons of conflicts.
> 
> I am hoping those issues will be fixed in the original library so we can switch back to the same.

Agree 100%.

Reply:
> Also, I feel we will need to use exact/locked version of this library in `ably-ruby` to make sure it works as expected for provided fix. Otherwise, including relevant missing tests will be super useful.

I’m afraid I don’t understand at all what you mean here. Why do we need to use a locked version? What do you mean "including relevant missing tests"?

Reply:
I meant using exact version for `ably-em-http-request`. e.g `1.1.7` instead of `~1.1`.
Or if we can add a integration test for the updated PR to make sure it won't break for future changes when using `~1.1`.

Reply:
Anyways, since library is not actively maintained, I don't see changes happening anytime soon !
Or we totally get rid of eventmachine in the future : )

Reply:
> I meant using exact version for `ably-em-http-request`. e.g `1.1.7` instead of `~1.1` in `ably-ruby` Or if we can add a integration test for the updated PR to make sure it won't break for future changes when using `~1.1`.

Ah OK, we don't need to lock it, we can just write (assuming the release that fixes this bug is 1.1.8) `[">= 1.1.8", "< 2.0.0"]`, right?

Reply:
> Anyways, since library is not actively maintained, I don't see changes happening anytime soon ! Or we totally get rid of eventmachine in the future : )

Yeah, I think getting rid of EventMachine is the best long-term idea (see https://github.com/ably/ably-ruby/issues/189). As mentioned in this PR, its handling is TLS is quite poor.

Reply:
Yeah, something like that. I think ~1.1 should resolve to the same.

Reply:
I don't think so — per [this documentation](https://bundler.io/guides/gemfile.html) `~> 1.1` is identical to "`>= 1.1` and `< 2.0`"  (i.e. doesn't guarantee that you'll get 1.1.8 or above).

Reply:
Which version actually gets chosen is up to the person / tool (e.g. Bundler) who installs the gem. But basically, yes.

Reply:
Okay, I will resolve this one for now 👍